src/faults/AbstractPerturbation.jl

```diff
@@ -48,21 +48,26 @@ function simulate(np::AbstractPerturbation, powergrid::PowerGrid, x1::Array, tim
     @assert np.tspan_fault[2] <= last(timespan) "fault cannot end in the future"
 
     np_powergrid = np(powergrid)
+    regular = rhs(powergrid)
+    error = rhs(np_powergrid)
+    _f = (dx, x, p, t) -> p ? regular(dx,x,p,t) : error(dx,x,p,t)
```

Comment:
```suggestion
    _f = (dx, x, p, t) -> p ? regular(dx,x,nothing,t) : error(dx,x,nothing,t)
```
I wouldn't pass on the parameter since it is more of an implementation detail.